notifier/notifier.go

```diff
@@ -383,25 +391,25 @@ func (n *Notifier) Stop() {
 
 // Describe implements prometheus.Collector.
 func (n *Notifier) Describe(ch chan<- *prometheus.Desc) {
```

Comment:
Yep, except how to address the data accessibility problem. Should I add the metrics as a field in AMSet too?

Also we have been doing this global pattern with registration in `init()` in a lot of places, should I open an issue to clean it up?